### ISSUE TITLE
False positive in unnecessary_nesting_linter()

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#2245]
Found while linting {potools}:

https://github.com/MichaelChirico/potools/blob/05e873dade3e3d148af5dc4a5b0c5206e6e511b9/R/specials_metadata.R#L68-L79C4

```r
lint(
  text = readLines("https://raw.githubusercontent.com/MichaelChirico/potools/3eaf727e03b1238d01492b99f58f938587b4b73b/R/specials_metadata.R"),
  linters = unnecessary_nested_if_linter()
)
```

**Comment by MichaelChirico** [comment]
Here's a reprex:

```r
lint(text = "if (A) { b=c; if (D) { e } }", linters = unnecessary_nesting_linter())
```

The key is that `=` is used, so `<expr>` is not its parent.
